File: src/main/java/com/taxidriverhk/hkadbus2/activity/filter/LoggingFilter.java
package com.taxidriverhk.hkadbus2.activity.filter;

import com.google.inject.Singleton;
import org.apache.logging.log4j.ThreadContext;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import java.io.IOException;
import java.util.UUID;

@Singleton
public class LoggingFilter implements Filter {

    private static final String REQUEST_ID = "requestId";

    @Override
    public void init(final FilterConfig filterConfig) throws ServletException {}

    @Override
    public void doFilter(final ServletRequest servletRequest, final ServletResponse servletResponse, final FilterChain filterChain)
            throws IOException, ServletException {
        ThreadContext.put(REQUEST_ID, UUID.randomUUID().toString());
        filterChain.doFilter(servletRequest, servletResponse);
    }

    @Override
    public void destroy() {}
}

File: src/main/kotlin/com/taxidriverhk/hkadbus2/activity/filter/LoggingFilter.kt
package com.taxidriverhk.hkadbus2.activity.filter

import com.google.inject.Singleton
import org.apache.logging.log4j.ThreadContext
import org.apache.logging.log4j.kotlin.Logging

import javax.servlet.Filter
import javax.servlet.FilterChain
import javax.servlet.FilterConfig
import javax.servlet.ServletRequest
import javax.servlet.ServletResponse
import java.util.UUID

@Singleton
class LoggingFilter : Filter {

    companion object : Logging {

        private val REQUEST_ID = "requestId"
    }

    override fun init(filterConfig: FilterConfig) {}

    override fun doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain) {
        ThreadContext.put(REQUEST_ID, UUID.randomUUID().toString())
        filterChain.doFilter(servletRequest, servletResponse)
    }

    override fun destroy() {}
}
